rcp/org.testeditor.web.testexecutionlogview/app/testLog.service.ts

```diff
@@ -8,36 +8,52 @@ import { TestExecutionLog } from "./model";
 export class TestLogService {
 
     port: String
+    knownTestruns: String[]
+    public testExecutionLogsChanged = new EventEmitter<Object>()
 
     constructor(private http: Http) {
         this.port = window.location.port
         // Use this for local setup with nodejs for development cycle
-        // this.port = "19091"
+        //this.port = "19091"
     }
 
     getTestExecutionLogs(): Promise<TestExecutionLog[]> {
-        return this.http.get('http://localhost:' + this.port + '/services/testruns').toPromise().then(response =>
+        let result = this.http.get('http://localhost:' + this.port + '/services/testruns').toPromise().then(response =>
             response.json().entries as TestExecutionLog[]
-        ).catch(this.handleError);
+        ).catch(this.handleError)
+        result.then(testExecLogs => {
+            this.knownTestruns = []
+            for (let run of testExecLogs) {
+                this.knownTestruns.unshift(run.name)
+            }
+        })
+        return result
     }
 
     getTestExecutionLogWithContent(filename: String): Promise<TestExecutionLog> {
-        return this.http.get('http://localhost:' + this.port + '/services/testruns/' + filename + "/logGroups").toPromise().then(response =>
+        let result = this.http.get('http://localhost:' + this.port + '/services/testruns/' + filename + "/logGroups").toPromise().then(response =>
             response.json() as TestExecutionLog
         ).catch(this.handleError);
+        result.then(log => {
+            if (!(this.knownTestruns.indexOf(log.name) > -1)) {
```

Comment:
this.knownTestruns.indexOf(log.name) < 0 would be more readable, what do you think?
!this.knownTestruns.contains(log.name) would be even better, but I don't know whether "contains" exists ...